### PR TITLE
Enhancing User Interface and Experience: Version Updates, Layout Improvements, and Detailed Form Revisions

### PR DESCRIPTION
**Pull request by kpqdap**
This pull request includes several changes to improve the user interface, update the environment variable for system prompts, and upgrade the application version. The main changes are:

User interface improvements:

* [`src/app/settings/layout.tsx`](diffhunk://#diff-2e44d895401410ed8d96840223f63de61d4a278e2bcdc1c2fba8d64b84db99eeL17-L24): Modified the layout of the settings menu to improve the user experience. The changes include adjusting the grid size and overflow properties.
* `src/features/settings/settings-menu-items.tsx` and `src/features/settings/settings-menu/settings-menu-items.tsx`: Updated the text and icons in the settings menu items for better clarity and consistency. [[1]](diffhunk://#diff-644336b5c0774bf753e68496b61bd9cd2e4a6caaba623a702dd2a0aac1de5dbfL1-R11) [[2]](diffhunk://#diff-01daabf4a250b3c5d4a972b5a314383d40fc4c703c1419023b10333d38a0fb33L1-R10)
* `src/features/settings/tenant-details.form.tsx` and `src/features/settings/user-details-form.tsx`: Updated the forms in the settings page to improve the user experience. The changes include modifying the layout, updating the text, and adding a new context prompt. [[1]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L8-R10) [[2]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L68-R113) [[3]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L85-R123) [[4]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L97-R133) [[5]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L109-R146) [[6]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L124-L156) F04a9306L67R122, F04a9306L55R109, [[7]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243L9-R9) [[8]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243R23-R44) [[9]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243R60-R96) [[10]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243R112) [[11]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243L70-R194)

Environment variable update:

* `src/.env.example` and `src/features/chat/chat-services/chat-api-helper.ts`: Updated the environment variable for system prompts from `SYSTEM_PROMPT` to `NEXT_PUBLIC_SYSTEM_PROMPT`. [[1]](diffhunk://#diff-37d244a68001b3a8ce7d98f05894470bb8e6fb387402adc29351d41570e3e1e6R48-L54) [[2]](diffhunk://#diff-6d77056e441a1e3e83af3fb3c1d13c45e30a871dc6fe39be966b224dd004d3ffL18-R18) [[3]](diffhunk://#diff-6d77056e441a1e3e83af3fb3c1d13c45e30a871dc6fe39be966b224dd004d3ffL33-R33)

Application version upgrade:

* [`src/app-global.ts`](diffhunk://#diff-6f6b4494ef9199e92a8afda39fbb184b4e854549a8dd44d0706e132a2b8fc3f2L1-R1): Upgraded the application version from "1.3.8" to "1.4.0".This pull request primarily focuses on refining the user interface and improving the user experience. The changes include updates to the application version, modifications to the layout, changes to the settings menu, and enhancements to the user and tenant details forms. 

User Interface and Experience Improvements:

* [`src/app-global.ts`](diffhunk://#diff-6f6b4494ef9199e92a8afda39fbb184b4e854549a8dd44d0706e132a2b8fc3f2L1-R1): Updated the application version from 1.3.8 to 1.4.0.
* [`src/app/settings/layout.tsx`](diffhunk://#diff-2e44d895401410ed8d96840223f63de61d4a278e2bcdc1c2fba8d64b84db99eeL17-L24): Adjusted the layout of the settings page for better user experience.

Settings Menu Changes:

* [`src/features/settings/settings-menu-items.tsx`](diffhunk://#diff-644336b5c0774bf753e68496b61bd9cd2e4a6caaba623a702dd2a0aac1de5dbfL1-R11): Updated the menu items in the settings, including changes to the text and the icons used. [[1]](diffhunk://#diff-644336b5c0774bf753e68496b61bd9cd2e4a6caaba623a702dd2a0aac1de5dbfL1-R11) [[2]](diffhunk://#diff-01daabf4a250b3c5d4a972b5a314383d40fc4c703c1419023b10333d38a0fb33L1-R10)
* [`src/features/settings/settings-menu.tsx`](diffhunk://#diff-086aec839c35f66153cc6c723ee8a72158a1211ff55cbe9afe8c6b111bf6b2e1L9-R9): Modified the settings menu layout.

User and Tenant Details Form Enhancements:

* [`src/features/settings/user-details-form.tsx`](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243R23-R44): Enhanced the user details form by providing more context for the user prompt and refining the layout. [[1]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243R23-R44) [[2]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243R60-R96) [[3]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243R112) [[4]](diffhunk://#diff-816aaeb71f1218485ea0ce05dbddc4d7ef3ac2d990669cb14cf199692a001243L70-R196)
* [`src/features/settings/tenant-details.form.tsx`](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L8-R10): Improved the tenant details form by reorganizing the layout and providing more context for the tenant prompt. [[1]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L8-R10) [[2]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L68-R113) [[3]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L85-R123) [[4]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L97-R133) [[5]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L109-R146) [[6]](diffhunk://#diff-bdc818cd70309bb77a55ac3049efd38aee19d3860e9838e0e0cb3af5f7fe32c7L124-L156) F04a9306L67R122)

New Features:

* [`src/app/api/prompt/route.ts`](diffhunk://#diff-cbaff1594db3baab9bbf9e1eda542b179d2751bb5a9d8cfc6e44c13faf20b043R1-R6): Added a new route to get the system prompt.
* [`src/features/theme/readable-systemprompt.tsx`](diffhunk://#diff-2a06f81226b87503fccbc434fff63030581bf2506af367490b4914910a155eacR1-R27): Created a new component to display the system prompt.
* [`src/middleware.ts`](diffhunk://#diff-0626369278c89781505d2c04865ecce1302e979271bcfb471e029c19addc13a0R45): Added a new path to the middleware configuration for the system prompt API.…settings-menu.tsx, settings-menu-items.tsx, settings-menu-items.tsx, layout.tsx

### CODE REVIEW

File: src/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "src",
-  "version": "1.4.8",
+  "version": "1.4.0",
```
Comment (QDAP-Fred) [inline src/package.json:3]: it's decreasing! ;)

File: src/app/api/prompt/route.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export async function GET(): Promise<Response> {
```
Comment (QDAP-Fred) [inline src/app/api/prompt/route.ts:1]: that feels overkill, you could just turn SYSTEM_PROMPT env var into a public variable so you can access it directly from client components

**Review by QDAP-Fred — DISMISSED** [review]

File: src/features/settings/settings-menu/settings-menu-items.tsx [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 import { isTenantAdmin } from "@/features/auth/helpers"
 
 const menuItems = [
-  { url: "/settings/details", icon: <UserRoundCog size={16} />, text: "Personal Details" },
-  { url: "/settings/tenant", icon: <Settings size={16} />, text: "Tenant Details", adminRequired: true },
-  { url: "/settings/details", icon: <FileClock size={16} />, text: "Chat History - Coming Soon" },
-  { url: "/settings/details", icon: <CircleHelp size={16} />, text: "Help & Support - Coming Soon" },
+  { url: "/settings/details", icon: <UserRoundCog size={16} />, text: "Your details" },
+  { url: "/settings/details", icon: <FileClock size={16} />, text: "Chat history - Coming soon" },
+  { url: "/settings/details", icon: <CircleHelp size={16} />, text: "Help & Support - Coming soon" },
```
[the remark this answers is not in the document]
Comment (kpqdap) [reply src/features/settings/settings-menu/settings-menu-items.tsx:9]: Just using it in place of temporary redirects

File: src/features/settings/user-details-form.tsx [hunk 4]
```diff
@@ -67,69 +122,78 @@ export const UserDetailsForm: React.FC<PromptFormProps> = () => {
...
-        <Typography variant="h5">
-          Email:
-          {isLoading ? (
-            <CardSkeleton />
-          ) : (
-            <div>
-              <b>{session?.user?.email}</b>
-            </div>
+        <div className="m-2">
+          <Typography variant="h5" className="flex items-center">
```
[the remark this answers is not in the document]
Comment (kpqdap) [reply src/features/settings/user-details-form.tsx:131]: thanks - all sorted

File: src/package.json (hunk 1 above)
Comment (kpqdap) [reply src/package.json:3]: haha yeah - figured users would get confused

File: src/features/settings/user-details-form.tsx [hunk 5]
```diff
@@ -67,69 +91,76 @@ export const UserDetailsForm: React.FC<PromptFormProps> = () => {
...
+        </Form.Field>
+
+        {!isLoading && (
+          <div className="flex justify-end">
+            <Form.Submit asChild>
+              <Button type="submit" variant="default" disabled={isSubmitting} className="mr-2">
+                {isSubmitting ? "Updating..." : "Update"}
+              </Button>
+            </Form.Submit>
+            {/* <Button onClick={async e => handleSubmit(e, true)} variant="destructive" disabled={isSubmitting}>
```
Comment (qdap-bp) [inline src/features/settings/user-details-form.tsx:155]: does it need to be kept?

**Review by qdap-bp — APPROVED** [review]

**Review by QDAP-Fred — APPROVED** [review]

**Pull request merged**